fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanApplicationWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -991,10 +1101,19 @@ public CommandProcessingResult modifyApplication(final Long loanId, final JsonCo
                 existingLoanApplication.updateTransactionProcessingStrategy(strategy);
             }
 
-            final String collateralParamName = "collateral";
-            if (changes.containsKey(collateralParamName)) {
-                final Set<LoanCollateral> loanCollateral = this.loanCollateralAssembler.fromParsedJson(command.parsedJson());
-                existingLoanApplication.updateLoanCollateral(loanCollateral);
+            /**
+             * TODO: Allow other loan types.
+             */
```

Comment:
is this TODO done?

Reply:
Nope. This can be considered as a future work. In this context we'll allow this for the individual loan type only